example/lib/wait_for_email.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:example/constants.dart';
+import 'package:flutter/material.dart';
+
+class WaitForEmail extends StatelessWidget {
```

Comment:
Thanks for creating this page. Actually, I have updated the `onSuccess` call of `SupaMagicAuth` widget to be called once the user successfully authenticates, because I thought it might be better if the user didn't have to worry about listening to auth state to react to when the user comes back to the app by clicking the email link.

What do you think about introducing another parameter in `SupaMagicAuth` widget where developers can pass a widget to display once an email has been sent to the user. Something like this (we can probably come up with better name than `emailSentWidget` 😂):

```dart
class SupaMagicAuth extends StatefulWidget {
  ...
  /// Widget to dipsplay to the user once an email has been sent to them
  final Widget? emailSentWidget;
  ...
  @override
  Widget build(BuildContext context) {
    return _hasSentEmail?
    Form(...):
    emailSentWidget == null?
    Text('Email sent'):
    emailSentWidget
```

Reply:
>onSuccess? should it be called onSignedIn (or a better name) then? I thought it was called on success of the operation ie sending the email. (like the other widgets).

I like the idea of renaming `onSuccess` to `onSignin`. Let's also rename the `onSuccess` on the signup related widgets to `onSignup` as well!

>emailSentWidget: I guess we could provide a default impl. It should probably also work for all the cases where we wait for something external. (email, sms, socials auth), and it should probably be possible to let the user decide what to do if he wants?

Yeah, this is a hard decision, isn't it. I personally think that people who are going to use this package are those who want to get something up and running ASAP, and if they really wanted the finest control over what happens, they would create their own signin/ signup form themselves. I personally would prefer this package to take care as much stuff around listening to auth state to sign in users and stuff. I like the style of just providing  (soon to be renamed)
 `onSuccess` method, which gets called when the user actually completes the sign in/ signup process. 

>We also wait for the authentication process to finish when doing social auth and phone_auth.

Yeah, I think I only covered magic link case, and haven't covered other use cases, so there definitely needs some polishing up. 

>so either we handle all those cases, or we let the user handles them by providing easy to use callbacks?

I think it's better to handle them for the users, but open to opinions!

>what do you think of having a onWaitForEmail (or onWaitForUserAction or ...) callback in the appropriate places?

Again, the same thing as above, but I want to provide a simple developer experience, where the developer doesn't have to worry about listening to auth state to complete the Sign-in process, so I don't know if introducing this callback is the ideal way, but always open to more opinioins!